### ISSUE TITLE
[Bug][Regression][32-33]: Dark theme is not applied to Dev Tools

### ISSUE THREAD

**Issue by AviVahl** [electron/electron#43367]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

32.0.0, 32.0.1, 32.0.2, 33.0.2, 33.1.0, 34.0.0-alpha.7

### What operating system(s) are you using?

Other Linux

### Operating System Version

Arch Linux with GNOME

### What arch are you using?

x64

### Last Known Working Electron version

31.4.0

### Expected Behavior

When opening dev tools, they should be dark themed, assuming Settings -> Theme is set to "Dark" or "Browser preference" (with the system being dark)

### Actual Behavior

Dev tools open in light theme, no matter what which setting is set.

### Additional Information

Setting the theme to Light and then back to Dark causes it to actually apply. That is until you close and re-open dev tools.

[Screencast from 2024-08-20 10-57-52.webm](https://github.com/user-attachments/assets/457c52f3-2e0f-4372-af2a-cba4f32774c1)

**Comment by AviVahl** [comment]
https://github.com/wixplosives/example-electron-app

**Comment by geedys** [comment]
This issue also occurs on the macOS platform. My macOS version: Sonoma 14.6.1 (23G93)

**Comment by nathanbabcock** [comment]
Also happens on both Windows 10 and 11 (so not a platform-specific issue it seems)

**Comment by Galkon** [comment]
Happening on macOS sonoma electron 32.0.1

**Comment by sparecycles** [comment]
From my experience (above duplicate)...

Reproducible in fiddle as well. versions `32.0.[01]` and `33.0.0-alpha.1`.

- Open dev tools 😎 
- Open dev tools settings😎 
- Set dark mode 🤓 
- Close dev tools
- Open dev tools. 😎!

**Comment by automationghost** [comment]
yes annoying please fix .. I updated from 30 to 32 and now have this issue

**Comment by cucbin** [comment]
Already checked which commit introduced the issue. Will submit a PR in further at the right time.

**Comment by TheOnlyOnlyName** [comment]
Hope this gets fixed, annoying to have to manually set it to dark every time I open devtools.

**Comment by Corvus278** [comment]
still happens with `32.1.0` :(

**Comment by danielweck** [comment]
> Edit created a quick workaround with this train of thought while they fix this 😎 evil bug. Use the devtools-opened event to toggle back and forth

Thank you for the tip :)

**Comment by TuKun33** [comment]
> I updated from 30 to 32 and now have this issue.. even though dark mode is set .. i have to toggle to light and back to dark so it is applied as the op already showed
> 
> Edit created a quick workaround with this train of thought while they fix this 😎 evil bug. Use the devtools-opened event to toggle back and forth
> 
> ```js
> import {app, BrowserWindow ,nativeTheme} from 'electron';
> 
> function toggleNativeTheme() {
>   // First, set the theme to 'light'
>   nativeTheme.themeSource = 'light';
> 
>   // After a short delay, set it to 'dark'
>   setTimeout(() => {
>     nativeTheme.themeSource = 'dark';
>   }, 100); // 100ms delay; adjust if necessary
> }
> 
> browserWindow?.webContents.openDevTools();
> browserWindow.webContents.on('devtools-opened', () => {
>   toggleNativeTheme();
> });
> ```

Thinks your code 👋🏼, But it's better to `on('devtools-opened')` before `openDevTools`
```javascript
window.webContents.on('devtools-opened', () => {})
window.webContents.openDevTools();
```

**Comment by ashgrover** [comment]
Can confirm this bug in 32.1.2

**Comment by AviVahl** [comment]
Issue still occurs with electron 33.0.0

**Comment by JawadAhbab** [comment]
My eyes are burning. Please resolve this issue ASAP. 🥹

**Comment by sparecycles** [comment]
confirmed fixed in `electron-nightly@35.0.0-nightly.20241115` 👍
